src/toxcore/onion/onion_client.rs

```diff
@@ -292,6 +294,24 @@ impl OnionClient {
         }
     }
 
+    fn is_redundant_ping(&self, pk: PublicKey, search_pk: PublicKey, request_queue: &RequestQueue<AnnounceRequestData>) -> bool {
+        let check_pks = |data: &AnnounceRequestData| -> bool {
+            let request_search_pk = if let Some(friend_pk) = data.friend_pk {
+                friend_pk
+            } else {
+                self.dht.pk
+            };
+            data.pk == pk && search_pk == request_search_pk
+        };
+        let request = request_queue.get_values()
+            .find(|(_ping_time, request_data)| check_pks(request_data));
+        if let Some((ping_time, _request_data)) = request {
```

Comment:
Now the check for timeout is redundant here, check with is_some

Reply:
As I know, the timeout of RequestQueue is the timeout of Queue itself and this check is for preventing redundant ping.

Reply:
RequestQueue timeout is ANNOUNCE_TIMEOUT which is 300 seconds and cheking timeout is 10 seconds.